### PR TITLE
update some docs

### PR DESCRIPTION
**Pull request by spencersalazar**

### CODE REVIEW

File: src/chuck_lang.cpp [hunk 1]
```diff
@@ -1491,72 +1495,90 @@ static t_CKUINT MidiOut_offset_data = 0;
...
 
     // add member variable
-    MidiMsg_offset_data3 = type_engine_import_mvar( env, "int", "data3", FALSE );
+    doc = "Third byte of a Midi message, usually a velocity value.";
+    MidiMsg_offset_data3 = type_engine_import_mvar( env, "int", "data3", FALSE, doc.c_str() );
     if( MidiMsg_offset_data3 == CK_INVALID_OFFSET ) goto error;
 
     // add member variable
-    MidiMsg_offset_when = type_engine_import_mvar( env, "dur", "when", FALSE );
+    doc = "Time when the MidiMsg occured, relative to the start of the file (only valid for MidiFileIn).";
```
Comment (nshaheed) [inline src/chuck_lang.cpp:1524]: occurred

File: src/ugen_xxx.cpp [hunk 2]
```diff
@@ -970,258 +972,329 @@ DLL_QUERY lisa_query( Chuck_DL_Query * QUERY )
...
 
     //---------------------------------------------------------------------
     // init class: LiSa; overloaded class for both LiSaBasic and LiSaMulti
     //              - probably don't need the others anymore....
     // author: Dan Trueman (dan /at/ music.princeton.edu)
     //---------------------------------------------------------------------
 	
-    
+    doc = "LiSa provides basic live sampling functionality.\
+    An internal buffer stores samples chucked to LiSa's input.\
```
Comment (nshaheed) [inline src/ugen_xxx.cpp:984]: The spaces at the beginning of the lines are going to add a lot of odd spacing when this is printed out

File: src/ugen_xxx.cpp [hunk 3]
```diff
@@ -970,258 +972,329 @@ DLL_QUERY lisa_query( Chuck_DL_Query * QUERY )
...
+    func->doc = "Return an available voice (one that is not currently playing). Return -1 if no voice is available.";
     if( !type_engine_import_mfun( env, func ) ) goto error;
     
     // set maxvoices
     func = make_new_mfun( "int", "maxVoices", LiSaMulti_ctrl_maxvoices );
+    func->doc = "Set the maximum number of voices allowable; 10 by default (200 is the current hardwired internal limit).";
     func->add_arg( "int", "val" );
     if( !type_engine_import_mfun( env, func ) ) goto error;
     func = make_new_mfun( "int", "maxVoices", LiSaMulti_cget_maxvoices);
+    func->doc = "Get the maximum number of voices allowable; 10 by default (200 is the current hardwired internal limit).";
```
Comment (nshaheed) [inline src/ugen_xxx.cpp:1235]: 256

File: src/ugen_xxx.cpp [hunk 4]
```diff
@@ -970,258 +972,329 @@ DLL_QUERY lisa_query( Chuck_DL_Query * QUERY )
...
     if( !type_engine_import_mfun( env, func ) ) goto error;
     
     // get free voice
     func = make_new_mfun( "int", "getVoice", LiSaMulti_cget_voice );
+    func->doc = "Return an available voice (one that is not currently playing). Return -1 if no voice is available.";
     if( !type_engine_import_mfun( env, func ) ) goto error;
     
     // set maxvoices
     func = make_new_mfun( "int", "maxVoices", LiSaMulti_ctrl_maxvoices );
+    func->doc = "Set the maximum number of voices allowable; 10 by default (200 is the current hardwired internal limit).";
```
Comment (nshaheed) [inline src/ugen_xxx.cpp:1231]: This is now 256 https://github.com/ccrma/chuck/blob/949b7e8472a746820dbef71d93a6d0b144d59287/src/core/ugen_xxx.cpp#L3989

File: src/ulib_math.cpp [hunk 5]
```diff
@@ -75,142 +75,175 @@ DLL_QUERY libmath_query( Chuck_DL_Query * QUERY )
 
     // add class
     QUERY->begin_class( QUERY, "Math", "Object" );
+    QUERY->doc_class( QUERY, "Math contains the standard math functions, including all trignometric functions expects angles to be in radians, as well as random number generators." );
```
Comment (nshaheed) [inline src/ulib_math.cpp:78]: the phrasing on the trig functions is kinda wonky

File: src/ulib_math.cpp [hunk 6]
```diff
@@ -238,121 +271,140 @@ DLL_QUERY libmath_query( Chuck_DL_Query * QUERY )
...
     
     // add srandom (1.3.1.0)
     QUERY->add_sfun( QUERY, srandom_impl, "void", "srandom" );
     QUERY->add_arg( QUERY, "int", "seed" );
+    QUERY->doc_func( QUERY, "Provide a seed to the random function. Different seeds will generate very different sequences of random numbers even if the seeds are close together. " "Alternatively, a sequence of random numbers can be repeated by setting the same seed.");
     
     // go ahead and seed (the code can seed again for repeatability; 1.3.1.0)
     srandom( time( NULL ) );
-
+
```
Comment (nshaheed) [inline src/ulib_math.cpp:360]: remove

**Review by nshaheed — COMMENTED** [review]
Yay thanks for the doc updates!

File: src/ulib_math.cpp (hunk 5 above)
Comment (spencersalazar) [inline src/ulib_math.cpp:78]:
```suggestion
    QUERY->doc_class( QUERY, "Math contains the standard math functions, including trigonometric functions and random number generators. Trigonometric functions expect angle values to be in radians." );
```

File: src/ulib_math.cpp (hunk 6 above)
Comment (spencersalazar) [reply src/ulib_math.cpp:360]:
```suggestion

```

**Pull request merged**